batch/batches/ValidateLiveMediaServers/KAsyncValidateLiveMediaServers.class.php

```diff
@@ -22,83 +25,281 @@ public static function getType()
+			}
+
+			$entryServerNodePager->pageIndex++;
+			$entryServerNodes = self::$kClient->entryServerNode->listAction($entryServerNodeFilter, $entryServerNodePager);
+		}
+	}
+
+	protected function validateSimuliveEntryServerNode()
+	{
+		// getting simulive serverNode
+		$simuliveServerNode = KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'getSimuliveServerNode'), array());
+		if (!$simuliveServerNode)
+		{
+			KalturaLog::info('no simulive server node found');
+			return;
+		}
+
+		//getting all esn of the simulive serverNode
+		$entryServerNodes = self::getEntryServerNodes($simuliveServerNode->id);
+
+		// getting currently live events
+		$currentlyLiveEvents = self::getSimulivePlayableEvents();
+
+		// map between templateEntryId to eventId
+		$eventsMap = self::arrayColumn($currentlyLiveEvents, 'templateEntryId', 'id');
+		// map between entryId to entryServerNode object
+		$entryServerNodesMap = self::arrayColumn($entryServerNodes, 'entryId');
+
+		foreach ($eventsMap as $templateEntryId => $eventId)
+		{
+			// event exist but there's no entry serverNode - create the entryServerNode (only for simulve - has sourceEntryId)
+			if ($templateEntryId && !array_key_exists($templateEntryId, $entryServerNodesMap))
+			{
+				KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'registerSimuliveMediaServer'), array($templateEntryId));
+			}
+		}
+
+		foreach ($entryServerNodesMap as $liveEntryId => $entryServerNode)
+		{
+			// entryServerNode exist but there's no event - delete the entryServerNode
+			if (!array_key_exists($liveEntryId, $eventsMap))
+			{
+				/* @var $entryServerNode KalturaEntryServerNode */
+				KBatchBase::tryExecuteApiCall(array('KAsyncValidateLiveMediaServers', 'validateRegisteredEntryServerNode'), array($entryServerNode->partnerId, $entryServerNode->id));
+			}
+		}
+	}
+
+	/**
+	 * Return filter for live entry server nodes
+	 * @return KalturaEntryServerNodeFilter
+	 */
+	protected function getLiveEntryServerNodesFilter()
+	{
+		$entryServerNodeMinCreationTime = $this->getAdditionalParams("minCreationTime");
+		if(!$entryServerNodeMinCreationTime)
+		{
+			$entryServerNodeMinCreationTime = self::ENTRY_SERVER_NODE_MIN_CREATION_TIME;
+		}
+
+		$entryServerNodeFilter = new KalturaEntryServerNodeFilter();
+		$entryServerNodeFilter->orderBy = KalturaEntryServerNodeOrderBy::CREATED_AT_ASC;
+		$entryServerNodeFilter->createdAtLessThanOrEqual = time() - $entryServerNodeMinCreationTime;
+
+		$excludeServerIds = $this->getExcludeServerIds();
+		if ($excludeServerIds)
+		{
+			$entryServerNodeFilter->serverNodeIdNotIn = implode(',', $excludeServerIds);
+		}
+		return $entryServerNodeFilter;
+	}
+
+	/**
+	 * Return LiveStreamScheduleEvent filter
+	 * @return KalturaLiveStreamScheduleEventFilter
+	 */
+	protected static function getSimuliveEventsFilter()
+	{
+		$now = time();
+		$scheduleEventFilter = new KalturaLiveStreamScheduleEventFilter();
```

Comment:
Are we sure this filter won't get also the EntryScheduleEvent? (none live)

Reply:
Yes, it returns only events of type "KalturaLiveStreamScheduleEvent"